test/unit/math/mix/fun/complex_schur_decompose_test.cpp

```diff
@@ -0,0 +1,94 @@
+#include <test/unit/math/test_ad.hpp>
+#include <stdexcept>
+
```

Comment:
I don't see any tests in here that test whether things are getting doubly allocated or recomputed.  Are we just supposed to do that by inspection?

Reply:
That is done through the expression testing framework. The code to do so is in test/expressions

The basic outline is code is generated for each signature listed by `stanc --dump-stan-math-signatures`. This obviously requires the signature to be exposed before being tested, but you can also run it manually; I have